components/FormularAddRecipe.js

```diff
@@ -0,0 +1,197 @@
+
+const StyledButton = styled.button`
+  background-color: hotpink;
+  width: 480px;
+  margin: 10px;
+  height: 40px;
+  font-size: 18px;
+  border-radius: 5px;
+  align-item: center;
+`;
+
+export default function RecipeForm({ onSubmit, formName, defaultData }) {
+  const [ingredients, setIngredients] = useState([]);
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+    onSubmit({ ...data, ingredients });
+  }
+
+  return (
+    <StyledDiv>
+      <StyledForm
+        aria-labelledby={formName}
+        id="recipeForm"
+        onSubmit={handleSubmit}
+      >
+        <StyledInput>
+          <label htmlFor="title">Title</label>
+          <input
+            id="title"
+            name="title"
+            type="text"
+            defaultValue={defaultData?.title}
+          />
+        </StyledInput>
+        <StyledInput>
+          <label htmlFor="preparation">
+            preparation:
+            <select name="preparation" defaultValue="Oven">
+              <option value="Oven">Oven</option>
+              <option value="Microwave">Microwave</option>
+              <option value="Stove">Stove</option>
+              <option value="Grill">Grill</option>
+            </select>
+          </label>
+        </StyledInput>
+        <StyledInput>
+          <label htmlFor="course" defaultValue="Cake">
+            Course:
+            <select name="course" defaultValue="Cake">
+              <option value="Cake">Cake</option>
+              <option value="Dish">Dish</option>
+              <option value="Soup">Soup</option>
+              <option value="Salad">Salad</option>
+              <option value="Cookies">Cookies</option>
+              <option value="Drinks">Drinks</option>
+              <option value="Fingerfood">Fingerfood</option>
+            </select>
+          </label>
+        </StyledInput>
+        <StyledInput>
+          <label htmlFor="time">Time:</label>
+          <input
+            id="time"
+            name="time"
+            type="text"
+            defaultValue={defaultData?.time}
+          />
+        </StyledInput>
+        <StyledInput>
+          <label htmlFor="servings">Servings:</label>
+          <input
+            defaultValue="1"
+            id="servings"
+            name="servings"
+            type="number"
+            min="1"
+          />
+        </StyledInput>
+        <StyledInput>
+          <label htmlFor="instruction">Instuction</label>
+          <textarea
+            name="instruction"
+            id="instruction"
+            cols="40"
+            rows="10"
+            defaultValue={defaultData?.instruction}
+          ></textarea>
+        </StyledInput>
+      </StyledForm>
+
+      <form
+        onSubmit={(event) => {
+          event.preventDefault();
+          setIngredients([
+            ...ingredients,
+            {
+              ingredientId: nanoid(),
```

Comment:
your database will create an id automatically, there is no need for you to create another one manually

Reply:
At this point we do not send the ingredients to our database. Without adding an id we got the note "each child in a list needs an unique key-prop". Later we want to be able the delete items within this formular.

Reply:
You could work with the name of the ingredient for the key prop.

You should only work on what is relevant for the current user story and not take features of an upcoming user story into consideration.